compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/FakeOverrideGenerator.kt

```diff
@@ -92,10 +92,13 @@ class FakeOverrideGenerator(
                     FirTypeScope::getDirectOverriddenFunctions,
                     useSiteMemberScope,
                 )
+                if (updated) {
```

Comment:
Unfortunately this does not do what you probably expect. This is only a return from lambda, no more. It does not stop `processFunctionsByName` itself, sorry.